go/vt/vtctl/reparentutil/planned_reparenter.go

```diff
@@ -688,6 +672,20 @@ func (pr *PlannedReparenter) reparentTablets(
 		}(alias, tabletInfo.Tablet)
 	}
 
+	// If `PromoteReplica` call is requried, we should call it and use the position that it returns.
+	if promoteReplicaRequired {
```

Comment:
it would be kinda neat if `PromoteReplica` was a no-op if the tablet was already type:PRIMARY since it would simplify the logic here a bit (however, that would make the actual server-side implementation more involved, and possibly too much, so 🤷 )

Reply:
`PromoteReplica` is already idempotent. We can call it over and over and it won't be wrong per se, but we still only want to call it when required.

Also, I wanted to reduce the change to the `PRS` flow as much as possible and only make the small change of running `PromoteReplica` in parallel instead of sequentially.